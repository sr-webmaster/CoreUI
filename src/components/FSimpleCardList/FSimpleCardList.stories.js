import { MAIN } from '../../../.storybook/categories'
import FSimpleCardList from './FSimpleCardList.vue'

export default {
  title: `${MAIN}|FSimpleCardList`,
  id: 'FSimpleCardList'
}

/**
 * https://storybook.js.org/docs/guides/guide-vue/#step-4-write-your-stories
 */
export const Single = () => ({
  components: { FSimpleCardList },
  data: () => ({
    items: [
      { header: 'TOTAL OPPORTUNITIES', value: '0000' }
    ]
  }),
  template: `
    <v-container>
      <f-simple-card-list :items="items"/>
    </v-container>
  `
})

export const FourItems = () => ({
  components: { FSimpleCardList },
  data: () => ({
    items: [
      { header: 'TOTAL OPPORTUNITIES', value: '0000' },
      { header: '% IN PROGRESS', value: '00%' },
      { header: '% CLOSED WON', value: '00%' },
      { header: '% CLOSED LOST', value: '00%' }
    ]
  }),
  template: `
    <v-container>
      <f-simple-card-list :items="items"/>
    </v-container>
  `
})

export const Multiple = () => ({
  components: { FSimpleCardList },
  data: () => ({
    items: [
      { header: 'TOTAL OPPORTUNITIES', value: '0000' },
      { header: '% IN PROGRESS', value: '00%' },
      { header: '% CLOSED WON', value: '00%' },
      { header: '% CLOSED LOST', value: '00%' },
      { header: 'TOTAL OPPORTUNITIES', value: '0000' },
      { header: '% IN PROGRESS', value: '00%' },
      { header: '% CLOSED WON', value: '00%' },
      { header: '% CLOSED LOST', value: '00%' },
      { header: 'TOTAL OPPORTUNITIES', value: '0000' },
      { header: '% IN PROGRESS', value: '00%' },
      { header: '% CLOSED WON', value: '00%' },
      { header: '% CLOSED LOST', value: '00%' }
    ]
  }),
  template: `
    <v-container>
      <f-simple-card-list :items="items"/>
    </v-container>
  `
})
