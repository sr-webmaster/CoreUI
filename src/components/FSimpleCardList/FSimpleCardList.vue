<template>
  <v-container
    grid-list-sm
  >
    <v-layout
      row
      wrap
    >
      <v-flex
        v-for="(item, key) in items"
        :key="key"
        class="pa-2"
      >
        <v-card
          color="grey lighten-2"
          class="pa-2 text-xs-center"
          center
        >
          <h6 class="font-weight-bold">
            {{ item.header }}
          </h6>
          <v-divider light />
          <h1>
            {{ item.value }}
          </h1>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
export default {
  name: 'SimpleCardList',
  props: {
    items: {
      type: Array,
      required: true
    }
  }
}
</script>
