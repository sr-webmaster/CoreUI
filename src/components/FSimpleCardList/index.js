import Component from './FSimpleCardList'
export default Component
