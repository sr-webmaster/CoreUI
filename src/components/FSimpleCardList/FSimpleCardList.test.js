import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import { FourItems } from './FSimpleCardList.stories'

describe('FSimpleCardList', () => {
  describe('Computed', () => {
    // Using shallowMount test the computed properties
  })

  describe('Methods', () => {
    // Using shallowMount test the methods
  })

  describe('Visuals', () => {
    test('Items', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(FourItems(), {
        localVue: createLocalVue().vue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.findAll('.v-card')).toHaveLength(4)
      expect(wrapper.text()).toContain('TOTAL OPPORTUNITIES')
      expect(wrapper.text()).toContain('% IN PROGRESS')
      expect(wrapper.text()).toContain('% CLOSED WON')
      expect(wrapper.text()).toContain('% CLOSED LOST')
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
