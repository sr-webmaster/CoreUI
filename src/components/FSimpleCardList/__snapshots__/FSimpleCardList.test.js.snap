// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FSimpleCardList Visuals Items 1`] = `
<div
  class="container"
>
  <div
    class="container grid-list-sm"
  >
    <div
      class="layout row wrap"
    >
      <div
        class="flex pa-2"
      >
        <div
          center=""
          class="pa-2 text-xs-center v-card v-sheet theme--light grey lighten-2"
        >
          <h6
            class="font-weight-bold"
          >
            
          TOTAL OPPORTUNITIES
        
          </h6>
           
          <hr
            class="v-divider theme--light"
          />
           
          <h1>
            
          0000
        
          </h1>
        </div>
      </div>
      <div
        class="flex pa-2"
      >
        <div
          center=""
          class="pa-2 text-xs-center v-card v-sheet theme--light grey lighten-2"
        >
          <h6
            class="font-weight-bold"
          >
            
          % IN PROGRESS
        
          </h6>
           
          <hr
            class="v-divider theme--light"
          />
           
          <h1>
            
          00%
        
          </h1>
        </div>
      </div>
      <div
        class="flex pa-2"
      >
        <div
          center=""
          class="pa-2 text-xs-center v-card v-sheet theme--light grey lighten-2"
        >
          <h6
            class="font-weight-bold"
          >
            
          % CLOSED WON
        
          </h6>
           
          <hr
            class="v-divider theme--light"
          />
           
          <h1>
            
          00%
        
          </h1>
        </div>
      </div>
      <div
        class="flex pa-2"
      >
        <div
          center=""
          class="pa-2 text-xs-center v-card v-sheet theme--light grey lighten-2"
        >
          <h6
            class="font-weight-bold"
          >
            
          % CLOSED LOST
        
          </h6>
           
          <hr
            class="v-divider theme--light"
          />
           
          <h1>
            
          00%
        
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
